### ISSUE TITLE
Apple Silicon M2 Loss Goes Up After 1-2 (Few) Epochs but not on Google Colab

### ISSUE THREAD

**Issue by felixm3** [keras-team/keras#18370]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS Ventura 13.5.1
- TensorFlow installed from (source or binary): Source
- TensorFlow version (use command below): 2.12.0 (Google Colab), 2.12.1 (Mac)
- Python version: 3.10.12 (Google Colab), 3.11.4 (Mac)
- GPU model and memory: T4 12GB RAM (Google Colab), Apple M2 Max (Mac)

**Describe the problem**.

I run this code on Mac (Apple Silicon M2 Max)

```
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Flatten

# Load the data
(x_train, y_train), _ = tf.keras.datasets.mnist.load_data()

# Normalize pixel values
x_train = x_train / 255.0

model = Sequential([
    Flatten(input_shape=(28, 28)),
    Dense(512, activation='relu'),
    Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

print("TensorFlow version: ", tf.__version__)

model.fit(x_train, y_train,
          epochs=10)
```

Loss immediately starts increasing after 1 epoch.

```
/Users/felixm/anaconda3/envs/tf230808/bin/python /Users/felixm/PycharmProjects/TFExamStudy/Coursera/2309_scratch.py 
TensorFlow version:  2.12.1
2023-09-03 16:53:30.575673: W tensorflow/tsl/platform/profile_utils/cpu_utils.cc:128] Failed to get CPU frequency: 0 Hz
Epoch 1/10
1875/1875 [==============================] - 7s 3ms/step - loss: 0.3915 - accuracy: 0.8903
Epoch 2/10
1875/1875 [==============================] - 7s 4ms/step - loss: 0.4136 - accuracy: 0.8952
Epoch 3/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.4905 - accuracy: 0.8897
Epoch 4/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.5585 - accuracy: 0.8878
Epoch 5/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.6251 - accuracy: 0.8857
Epoch 6/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.7148 - accuracy: 0.8848
Epoch 7/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.7854 - accuracy: 0.8822
Epoch 8/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.8779 - accuracy: 0.8816
Epoch 9/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.9782 - accuracy: 0.8821
Epoch 10/10
1875/1875 [==============================] - 6s 3ms/step - loss: 1.0523 - accuracy: 0.8798

Process finished with exit code 0

```

Running the same code on Google Colab, the loss never increases.

```

Downloading data from https://storage.googleapis.com/tensorflow/tf-keras-datasets/mnist.npz
11490434/11490434 [==============================] - 0s 0us/step
TensorFlow version:  2.12.0
Epoch 1/10
1875/1875 [==============================] - 11s 3ms/step - loss: 0.2004 - accuracy: 0.9412
Epoch 2/10
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0810 - accuracy: 0.9750
Epoch 3/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.0533 - accuracy: 0.9833
Epoch 4/10
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0363 - accuracy: 0.9883
Epoch 5/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.0274 - accuracy: 0.9911
Epoch 6/10
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0204 - accuracy: 0.9937
Epoch 7/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.0171 - accuracy: 0.9944
Epoch 8/10
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0131 - accuracy: 0.9955
Epoch 9/10
1875/1875 [==============================] - 5s 3ms/step - loss: 0.0123 - accuracy: 0.9961
Epoch 10/10
1875/1875 [==============================] - 6s 3ms/step - loss: 0.0114 - accuracy: 0.9960
<keras.callbacks.History at 0x780f114f2e60>

```

**Comment by m-kareem** [comment]
I am facing a similar issue with my Apple macBokAir M2

**Comment by tilakrayal** [comment]
@felixm3,
As mentioned, the loss was increasing on both v2.12 and v2.13, but whereas when we tried to execute the same code on tf-nightly the loss was reduced and also the accuracy also increased. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/0695aa7d67147ba4f6f6a135c33d8377/untitled1357.ipynb) and the MacOS screenshot for the reference.
<img width="1728" alt="image (19)" src="https://github.com/keras-team/keras/assets/81610181/2df0be66-97e3-48ab-9618-9b89613aeaa4">

Thank you!

**Comment by felixm3** [comment]
Hi @tilakrayal.

Is your recommendation to fix this issue to install TensorFlow version 2.15 then?

**Comment by tilakrayal** [comment]
@felixm3,
Yeah. It was an issue on Tensorflow v2.12 and v2.13, and the issue was **fixed on tf-nightly**, which will be added to the upcoming Tensorflow stable version release. Thank you!
